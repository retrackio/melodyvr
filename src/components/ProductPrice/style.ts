import styled from 'styled-components';

export const Container = styled.p`
  position: absolute;
  right: 0;
  top: 0;
  font-size: 1.4rem;
  padding: 5px 10px;
  background: rgba(255, 255, 255, 0.9);
`;
