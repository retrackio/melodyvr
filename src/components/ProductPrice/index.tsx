import React from 'react';
import * as Styled from './style';

interface Props {
  price: string | null;
  isFree: boolean;
}

function getPriceLabel({ price, isFree }: Props) {
  if (!isFree && !price) {
    return 'N/A';
  }
  if (isFree) {
    return 'FREE';
  }
  return <span>&pound;{price}</span>;
}

const ProductPrice = (props: Props) => (
  <Styled.Container>{getPriceLabel(props)}</Styled.Container>
);

export default ProductPrice;
