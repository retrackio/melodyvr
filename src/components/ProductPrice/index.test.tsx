import React from 'react';
import { shallow } from 'enzyme';
import ProductPrice from './';

describe('<ProductPrice />', () => {
  it('renders correctly when product is free', () => {
    const wrapper = shallow(<ProductPrice isFree={true} price={null} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when product is not free and no price is available', () => {
    const wrapper = shallow(<ProductPrice isFree={false} price={null} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when product is not free and a price is available', () => {
    const wrapper = shallow(<ProductPrice isFree={false} price="12.54" />);
    expect(wrapper).toMatchSnapshot();
  });
});
