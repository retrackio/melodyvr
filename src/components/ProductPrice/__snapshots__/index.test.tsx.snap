// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPrice /> renders correctly when product is free 1`] = `
<style__Container>
  FREE
</style__Container>
`;

exports[`<ProductPrice /> renders correctly when product is not free and a price is available 1`] = `
<style__Container>
  <span>
    Â£
    12.54
  </span>
</style__Container>
`;

exports[`<ProductPrice /> renders correctly when product is not free and no price is available 1`] = `
<style__Container>
  N/A
</style__Container>
`;
